### DISCUSSION TITLE
Memory allocation failed

### DISCUSSION THREAD

**Discussion by FireStone1029** [ImageMagick/ImageMagick#4999] [Help] [answered]
Im trying to convert a large svg 40,000x40,000 to a png and i keep getting this error  
memory allocation failed `Australia.svg' @ error/svg.c/RenderRSVGImage/565.

im using **magick convert -limit area 0 -define registry:temporary-path=D:/tmp Australia.svg -resize 8192x8192 Australia.png**
If some one can help me figure this out that would be much appreciated

 Extra Info
 Version: ImageMagick 7.1.0-28 Q16-HDRI x64 b83772f:20220320 https://imagemagick.org
 Width: 461.169PP
  Height: 461.169PP
  Area: 17.13GP
  List length: unlimited
  Memory: 7.97678GiB
  Map: 15.9536GiB
  Disk: unlimited
  File: 1536
  Thread: 8

**Comment by snibgo** [comment]
For IM v7, I suggest using just "magick", not "magick convert".

How much memory does your computer have? You have 1.6 billion pixels, which is a lot. It may work if you either (a) remove the limits on memory and map so all your memory can be used or (b) set them both to low numbers (eg 1 MB) to force IM to use disk.

IM uses one of three methods to rasterize SVG files: MVG, RSVG or Inkscape. Which one does your IM use? "-verbose" may tell you.

Do you have Inkscape? If not, I suggest you install it. Then IM should use it. Does that help? If not, try running Inkscape as a standalone program. Can that create the large PNG file?

**Reply by FireStone1029** [reply]
So i will try everything you said above & i do have inkscape and have tried loading it with it but its just loads for ever (i left it overnight so maybe like 9h and it was still loading)

**Reply by FireStone1029** [reply]
so i tryed
magick Australia.svg -define registry:temporary-path=D:/tmp Australia.tga
magick -define registry:temporary-path=D:/tmp Australia.svg -resize 8192x8192 Australia.png
magick -limit map 1GiB -limit memory 1GiB -define registry:temporary-path=D:/tmp Australia.svg -resize 8192x8192 Australia.png

and i have 16gib of ram
non of them worked i also could not get verbose working but im going to assume that im using inkscape 

if you want to give it a shot here is the file
https://drive.google.com/file/d/1dCx0sykd1KqC1DHJszVt4kEEjOM6-P4x/view?usp=sharing

**Comment by snibgo** [comment]
I don't seem to have a program that can read rar files. Can you zip it?

As you don't need all those 40,000x40,000 pixels, I suggest using a small "-density N" before reading the SVG. For example:
```
magick -verbose -limit area 0 -limit map 0 -define registry:temporary-path=D:/tmp -density 1 Australia.svg out.png
```

**Reply by FireStone1029** [reply]
https://drive.google.com/file/d/13U8e102Z8kG_G-zS3RSbpg-dHYg_rXOm/view?usp=sharing
Its now just a zip

on a side note i did get it to spit out a file by adding density 1 trying now at 76

**Comment by snibgo** [comment, marked as answer]
This works for me, on a 12GB Windows 8.1 laptop, 
```
magick -verbose -density 20 Australia.svg -resize 8192x8192 x.png
```
I get a warning from Inkscape:
```
** (process:1128): WARNING **: 00:11:46.058: CSS unrecoverable error

parsing error: 1:24:could not recognize next production
```
... but it creates the raster image.

The rasterized image is 8533x8533 pixels, which is slightly larger than you need, so we resize it.

If I omit "-density 20", the command takes longer, and creates a temporary file of 26 GB, before failing with:
```
magick: memory allocation failed `Australia.svg' @ error/svg.c/RenderRSVGImage/565.
```

If I insert "-limit memory 1MB -limit map 1MB" before reading the SVG, it still gives "memory allocation failed".

**Reply by FireStone1029** [reply]
so im getting the same thing, i did one with density at 72 and it work no issue it seems like it will fail if the file is larger then 35000x35000

thank you for you help

**Comment by fmw42** [comment]
There may be limits on the dimensions of the output for the format (PNG) that you are using. I am not PNG expert, so you should look into this.

**Reply by FireStone1029** [reply]
i did try tga as well as jpg and still got the same thing
